### PR TITLE
[WIP] pillow is not a hard dependency anymore

### PR DESCRIPTION
**Pull request by malev**
`pillow` you will only need pillow if you need to upload a thumbnail.
AS-784

### CODE REVIEW

File: binstar_client/utils/notebook/data_uri.py [hunk 1]
```diff
@@ -7,7 +7,11 @@
     from urlparse import urlparse
 except ImportError:
     from urllib.parse import urlparse
-from PIL import Image
+try:
+    from PIL import Image
+except ImportError:
+    from ...errors import PillowNotInstalled
+    raise PillowNotInstalled
```
Comment (srossross) [inline binstar_client/utils/notebook/data_uri.py]: @malev shouldn't this be called when PIL is used?

Comment (malev) [inline binstar_client/utils/notebook/data_uri.py]: you are right

Comment (malev) [inline binstar_client/utils/notebook/data_uri.py]: done

**Comment by srossross** [comment]
:+1:

**Pull request merged**